src/assets/Generator.Shared/OperationHelpers.cs

```diff
@@ -0,0 +1,368 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// This implements the ARM scenarios for LROs. It is highly recommended to read the ARM spec prior to modifying this code:
+    /// https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations
+    /// Other reference documents include:
+    /// https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation
+    /// https://github.com/Azure/adx-documentation-pr/blob/master/sdks/LRO/LRO_AzureSDK.md
+    /// </summary>
+    /// <typeparam name="T">The final result of the LRO.</typeparam>
+    internal class OperationHelpers<T>
```

Comment:
This is already a type with this name in Azure.Core shared https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Shared/OperationHelpers.cs. I think we should use a different name

Reply:
Unless it is expected to be shared, maybe it should have a mgmt prefix?

Reply:
do you have a recommended name? this is the one i had from our last conversation.